### PR TITLE
DevWs for TechniSat Digital GmbH containing containing TechniSat Shutter fixes

### PR DESCRIPTION
**Pull request by NArlt**

### CODE REVIEW

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+ */
+import groovy.json.JsonOutput
+
+metadata {
+	definition (name: "TechniSat Roller shutter switch", namespace: "TechniSat", author: "TechniSat", vid:"generic-shade",
+				mnmn: "SmartThings") {
+		capability "Window Shade"
+		capability "Window Shade Preset"
+		capability "Window Shade Level"
+        capability "Switch Level"
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy]: don't use both `"Switch Level"` and `"Window Shade Level"` and the same time

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+				range: it.range,
+				defaultValue: it.defaultValue,
+				required: false)
+		}
+	}
+}
+
+def installed() {
+	log.debug "installed()"
+    sendEvent(name: "supportedWindowShadeCommands", value: JsonOutput.toJson(["open", "close", "pause"]), displayed: false)
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy]: Wrong indentation.

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+		level = 100
+		shadeValue = "open"
+	} else if (cmd.value <= 0) {
+		level = 0 
+		shadeValue = "closed"
+	} else {
+		shadeValue = "partially open"
+	}
+	
+    def levelEvent = createEvent(name: "level", value: level, unit: "%")
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy]: If you want to send an event to `Window Shade Level` capability, use `"shadeLevel"` as event's name

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+}
+
+def ping() {
+	log.debug "ping()"
+	refresh()
+}
+
+def poll() {
+	sendHubCommand(refresh())
+}
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy]: This method is deprecated, it may be removed.

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]
I'm sorry, but I'll insist on creating a new vid, which would include Power and Energy Meters. If you'll have any troubles creating it, don't hesitate to contact us.

**Comment by NArlt** [comment]
Hi @PKacprowiczS ,
I've read the documentation and downloaded the SmartThings CLI. Unfortunately I don't know how to start. The documentation mentioned it is possible to create device-profiles with the DevWs, but this is not possible for hub-connected devices. So it seems I have to  start from scratch with SmartThings CLI. Is there a an example, tutorial or a step-by-step manual how to create the necessary structs for creating a vid for our device? I only worked with hub-connexted devices in the DevWs and therefor I'm not very experienced outside the DTH.

Thanks in advance.

BR NArlt

**Comment by PKacprowiczS** [comment]
@NArlt some insights about generating device presentation can be found here: https://community.smartthings.com/t/custom-capability-and-cli-developer-preview/197296 (start from `Generate / Post Device Configuration` section, since you don't use custom capabilities).

**Comment by NArlt** [comment]
@PKacprowiczS ,
thanks for the link!
With this tutorial I was able to create our own vid. The new view is atm in review. I will commit the changes as soon as it is approved by TechniSat.

**Comment by NArlt** [comment]
@PKacprowiczS ,
TechniSat approved the new view. But they decided to change the deviceJoinName back to 'TechniSat Roller Shutter Switch', which is the product name of this device. I hope this is acceptable for you.

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+		capability "Window Shade"
+		capability "Window Shade Preset"
+		capability "Window Shade Level"
+		capability "Power Meter"
+		capability "Energy Meter"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Configuration"     
+
+		fingerprint mfr: "0299", prod: "0005", model: "1A93", deviceJoinName: "TechniSat Roller Shutter Switch"
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy]: Please change name back.

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]
Unfortunately, all deviceJoinNames must follow same scheme, so it can't be `Roller Shutter Switch`.

**Review by PKacprowiczS — APPROVED** [review]
@greens @KKlimczukS

**Review by greens — APPROVED** [review]

**Comment by greens** [comment]
This will be in production on the 22nd (California time)

**Pull request merged**